src/policy/concrete.rs

```diff
@@ -128,6 +137,33 @@ impl fmt::Display for PolicyError {
 }
 
 impl<Pk: MiniscriptKey> Policy<Pk> {
+    /// Single-Node compilation
+    #[cfg(feature = "compiler")]
+    fn compile_huffman_taptree(policy: &Self) -> Result<TapTree<Pk>, Error> {
+        let compilation = policy.compile::<Tap>().unwrap();
+        Ok(TapTree::Leaf(Arc::new(compilation)))
+    }
+    /// Extract the Taproot internal_key from policy tree.
+    #[cfg(feature = "compiler")]
+    fn extract_key(policy: &Self, unspendable_key: Option<Pk>) -> Result<(Pk, &Policy<Pk>), Error> {
```

Comment:
This would require us to make the `compile_tr` function consume `self` which doesn't align with the `compile<Pk, Ctx>(&self)` function. I will update this function as suggested and use a cloned copy for the same until we discuss further.